AK/Platform.h

```diff
@@ -88,7 +88,13 @@
 #    define AK_OS_EMSCRIPTEN
 #endif
 
-#define ARCH(arch) (defined(AK_ARCH_##arch) && AK_ARCH_##arch)
+#define AK_MACRO_ERROR(...) (1 / 0)
```

Comment:
I don't know exactly how you'd get it in here, but the preprocessor has an `#error Message goes here` directive that would give a nicer message than 1/0 does.

Reply:
Regarding error message, it's quite clear (although, it would definitely benefit from `#error`):
```
/home/danklishch/code/serenity/Meta/Lagom/../../AK/Platform.h:91:32: error: division by zero in #if
   91 | #define AK_MACRO_ERROR(...) (1 / 0)
      |                                ^
/home/danklishch/code/serenity/Meta/Lagom/../../AK/Platform.h:97:102: note: in expansion of macro ‘AK_MACRO_ERROR’
   97 | #define ARCH(arch) (defined(AK_ARCH_##arch) && AK_ARCH_##arch) || (!defined(AK_VALID_ARCH_##arch) && AK_MACRO_ERROR(The specified architecture is nonexistent))
      |                                                                                                      ^~~~~~~~~~~~~~
/home/danklishch/code/serenity/Meta/Lagom/../../AK/Platform.h:201:26: note: in expansion of macro ‘ARCH’
  201 | #    if ARCH(AARCH64) || ARCH(x86_64)
      |                          ^~~~
```

Unfortunately, I think it is completely impossible to expand expression in `#if` to something containing `_Pragma` (in order to issue `#error`), 'cause [cppref](https://en.cppreference.com/w/cpp/preprocessor/conditional) says that the resulting condition should be calculated as an integral constant and the only functions available are `__has_include`, `defined` and `__has_cpp_attribute`. So, it looks like I'm limited to two almost identical options:

1) Use short-circuit evaluation to hide calculation of something illegal;
1) Make an inner define to expand to different values, so that it will produce illegal division on invalid arguments.

And choosing between these two, it seems like the first option makes code look a little bit easier to follow.

Reply:
Adding a flag macro is pretty common in other projects, where you expand a "flag" into a function-like macro. If that macro doesn't exist, you get a compile error. For example:
https://godbolt.org/z/z7KaTd556